cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -24,8 +24,27 @@ import (
 	"github.com/golang/glog"
 	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
 	"github.com/openebs/maya/pkg/util"
+	"github.com/pkg/errors"
 )
 
+var (
+	poolTypeMap    map[string]string
+	poolGroupByMap map[string]int
+	poolType       string
+	diskList       []string
+	diskCount      int
+	poolUID        string
+)
+
+func initCstorPoolVars(csp *apis.CStorPool) {
+	diskList = csp.Spec.Disks.DiskList
+	diskCount = len(diskList)
+	poolType = csp.Spec.PoolSpec.PoolType
+	poolUID = string(csp.ObjectMeta.UID)
+	poolTypeMap = map[string]string{"mirrored": "mirror", "raidz": "raidz", "raidz2": "raidz2"}
```

Comment:
make these as global constants

Reply:
We may not want to put CSP object fields in variables. What is making us to do so?
IMO -- only poolTypeMap  and poolGroupByMap should do the job.

Reply:
To reuse the same, I'm putting it in variables, otherwise I would have to call csp.something.something.something everywhere. Initializing the objects will reduce the overhead of calling the same again and again IMO.

Reply:
I would go by dot operators. Global things can cause problems. Anyway let us ask others opinions on this and then you can take a call.

Reply:
@AmitKumarDas commented about the same to reuse this. Not sure if i understood his comment correctly.https://github.com/openebs/maya/pull/893#discussion_r248940983

Reply:
Ok @utkarsh2102 . You can discuss and take a call. It is just a nit pick.